src/coverlet.core/Helpers/InstrumentationHelper.cs

```diff
@@ -442,7 +442,7 @@ private static bool IsTypeFilterMatch(string module, string type, string[] filte
         typePattern = WildcardToRegex(typePattern);
         modulePattern = WildcardToRegex(modulePattern);
 
-        if (new Regex(typePattern, s_regexOptions, TimeSpan.FromSeconds(10)).IsMatch(type) && new Regex(modulePattern, s_regexOptions, TimeSpan.FromSeconds(10)).IsMatch(module))
+        if (new Regex(typePattern).IsMatch(type) && new Regex(modulePattern).IsMatch(module))
```

Comment:
Can these type pattern regexes be pre-created and passed in instead of passing in strings? That's a lot of waste creating regexes every time a type is matched.

Reply:
This PR resolves the performance issue and we should limit the bug fix scope.

`IsTypeFilterMatch` is used by `IsTypeExcluded` and `IsTypeIncluded`. Introducing a local collection of exclude/include regex patterns might improve the performance. Please create an improvement issue.

Reply:
Looking at the .NET source code here: https://source.dot.net/#System.Text.RegularExpressions/System/Text/RegularExpressions/Regex.Match.cs,13

There's an internal RegexCache which is used when calling static methods of `Regex`, and not instantiating it. So an easy fix to try would be changing this line to:
```csharp
Regex.IsMatch(typePattern, type)
```

Reply:
Maybe this is not so easy. I tried to use the suggested `Regex.IsMatch(typePattern, type)` static regex method but unit tests and integration test are failing.

**Please create an separate improvement issue/PR. This PR should fix only the performance regression.**

Reply:
It failed since I wrote down the wrong order of the arguments: `IsMatch (string input, string pattern);`

Reply:
@xPaw The following unit tests are failing locally (Windows 11) but not on CI.
```
Coverlet.Core.Symbols.Tests.CecilSymbolHelperTests.GetBranchPoints_IgnoreBranchesIn_ScopedAwaitUsingStateMachine [FAIL]
Coverlet.Core.Symbols.Tests.CecilSymbolHelperTests.GetBranchPoints_IgnoreBranchesIn_AwaitUsingStateMachine [FAIL]
Coverlet.Core.Tests.CoverageTests.AwaitUsing [FAIL]
```

Reply:
Maybe because s_regexOptions was removed which specified multiline matching? You can still use that with Regex.IsMatch I think.

Reply:
I will check this in the next days ...

```
Starting test execution, please wait...
A total of 1 test files matched the specified pattern.
[xUnit.net 00:00:02.40]     Coverlet.Core.Symbols.Tests.CecilSymbolHelperTests.GetBranchPoints_IgnoreBranchesIn_ScopedAwaitUsingStateMachine [FAIL]
  Failed Coverlet.Core.Symbols.Tests.CecilSymbolHelperTests.GetBranchPoints_IgnoreBranchesIn_ScopedAwaitUsingStateMachine [12 ms]
  Error Message:
   Assert.Empty() Failure: Collection was not empty
Collection: [BranchPoint { Document = "C:\\GitHub\\coverlet\\test\\coverlet.core.tests\\S"···, EndOffset = 87, Offset = 85, OffsetPoints = [], Ordinal = 0, ··· }, BranchPoint { Document = "C:\\GitHub\\coverlet\\test\\coverlet.core.tests\\S"···, EndOffset = 192, Offset = 85, OffsetPoints = [], Ordinal = 1, ··· }]
  Stack Trace:
     at Coverlet.Core.Symbols.Tests.CecilSymbolHelperTests.GetBranchPoints_IgnoreBranchesIn_ScopedAwaitUsingStateMachine() in C:\GitHub\coverlet\test\coverlet.core.tests\Symbols\CecilSymbolHelperTests.cs:line 436
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[xUnit.net 00:00:02.44]     Coverlet.Core.Symbols.Tests.CecilSymbolHelperTests.GetBranchPoints_IgnoreBranchesIn_AwaitUsingStateMachine [FAIL]
  Failed Coverlet.Core.Symbols.Tests.CecilSymbolHelperTests.GetBranchPoints_IgnoreBranchesIn_AwaitUsingStateMachine [6 ms]
  Error Message:
   Assert.Empty() Failure: Collection was not empty
Collection: [BranchPoint { Document = "C:\\GitHub\\coverlet\\test\\coverlet.core.tests\\S"···, EndOffset = 80, Offset = 78, OffsetPoints = [], Ordinal = 0, ··· }, BranchPoint { Document = "C:\\GitHub\\coverlet\\test\\coverlet.core.tests\\S"···, EndOffset = 185, Offset = 78, OffsetPoints = [], Ordinal = 1, ··· }]
  Stack Trace:
     at Coverlet.Core.Symbols.Tests.CecilSymbolHelperTests.GetBranchPoints_IgnoreBranchesIn_AwaitUsingStateMachine() in C:\GitHub\coverlet\test\coverlet.core.tests\Symbols\CecilSymbolHelperTests.cs:line 420
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[xUnit.net 00:00:27.12]     Coverlet.Core.Tests.CoverageTests.Subtle_DoesNotReturnAttribute_InstrumentsCorrect [SKIP]
  Skipped Coverlet.Core.Tests.CoverageTests.Subtle_DoesNotReturnAttribute_InstrumentsCorrect [1 ms]
[xUnit.net 00:00:31.72]     Coverlet.Core.Tests.CoverageTests.If_DoesNotReturnAttribute_InstrumentsCorrect [SKIP]
  Skipped Coverlet.Core.Tests.CoverageTests.If_DoesNotReturnAttribute_InstrumentsCorrect [1 ms]
[xUnit.net 00:00:34.28]     Coverlet.Core.Tests.CoverageTests.AwaitUsing [FAIL]
  Failed Coverlet.Core.Tests.CoverageTests.AwaitUsing [902 ms]
  Error Message:
   Expected total branch is '0', actual '3'
  Stack Trace:
     at Coverlet.Core.Tests.TestInstrumentationAssert.ExpectedTotalNumberOfBranches(Document document, BuildConfiguration configuration, Int32 totalExpectedBranch) in C:\GitHub\coverlet\test\coverlet.core.tests\Coverage\InstrumenterHelper.Assertions.cs:line 137
   at Coverlet.Core.Tests.CoverageTests.AwaitUsing() in C:\GitHub\coverlet\test\coverlet.core.tests\Coverage\CoverageTests.AwaitUsing.cs:line 32
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
   at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
[xUnit.net 00:00:36.32]     Coverlet.Core.Tests.CoverageTests.AsyncAwait_Issue_669_2 [SKIP]
[xUnit.net 00:00:37.35]     Coverlet.Core.Tests.CoverageTests.UnreachableBranch_DoesNotReturnAttribute_InstrumentsCorrect [SKIP]
  Skipped Coverlet.Core.Tests.CoverageTests.AsyncAwait_Issue_669_2 [1 ms]
  Skipped Coverlet.Core.Tests.CoverageTests.UnreachableBranch_DoesNotReturnAttribute_InstrumentsCorrect [1 ms]
[xUnit.net 00:00:49.48]     Coverlet.Core.Tests.CoverageTests.Switch_DoesNotReturnAttribute_InstrumentsCorrect [SKIP]
  Skipped Coverlet.Core.Tests.CoverageTests.Switch_DoesNotReturnAttribute_InstrumentsCorrect [1 ms]
Results File: C:\GitHub\coverlet\artifacts/testresults/debug\TestResults-net8.0-coverlet.core.tests.trx
Html test results file : C:\GitHub\coverlet\artifacts/testresults/debug\TestResults-net8.0-coverlet.core.tests.html

Failed!  - Failed:     3, Passed:   254, Skipped:     5, Total:   262, Duration: 53 s - coverlet.core.tests.dll (net8.0) 
```

Reply:
The local unit test errors are observed after .NET SDK update to version 8.0.300 or 8.0.401 and therefore not related to  source code updated of this PR.